modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MysqlSchemaServerCodegen.java

```diff
@@ -0,0 +1,1028 @@
+                "VALUES", "VARBINARY", "VARCHAR", "VARCHARACTER", "VARYING", "VIRTUAL",
+                "WHEN", "WHERE", "WHILE", "WINDOW", "WITH", "WRITE",
+                "XOR",
+                "YEAR_MONTH",
+                "ZEROFILL"
+            )
+        );
+
+        // all types can be threaded as primitives except array, object and refs
+        languageSpecificPrimitives = new HashSet<String>(
+            Arrays.asList(
+                "bool",
+                "boolean",
+                "int",
+                "integer",
+                "double",
+                "float",
+                "string",
+                "date",
+                "Date",
+                "DateTime",
+                "long",
+                "short",
+                "char",
+                "ByteArray",
+                "binary",
+                "file",
+                "UUID",
+                "BigDecimal",
+                "mixed",
+                "number",
+                "void",
+                "byte"
+            )
+        );
+
+        // https://dev.mysql.com/doc/refman/8.0/en/data-types.html
+        typeMapping.put("array", "JSON");
+        typeMapping.put("map", "JSON");
+        typeMapping.put("List", "JSON");
+        typeMapping.put("boolean", "BOOL");
+        typeMapping.put("string", "TEXT");
+        typeMapping.put("int", "INT");
+        typeMapping.put("byte", "TEXT");
+        typeMapping.put("float", "DECIMAL");
+        typeMapping.put("number", "DECIMAL");
+        typeMapping.put("date", "DATE");
+        typeMapping.put("Date", "DATETIME");
+        typeMapping.put("DateTime", "DATETIME");
+        typeMapping.put("long", "BIGINT");
+        typeMapping.put("short", "SMALLINT");
+        typeMapping.put("char", "TEXT");
+        typeMapping.put("double", "DECIMAL");
+        typeMapping.put("object", "JSON");
+        typeMapping.put("integer", "INT");
+        typeMapping.put("ByteArray", "MEDIUMBLOB");
+        typeMapping.put("binary", "MEDIUMBLOB");
+        typeMapping.put("file", "MEDIUMBLOB");
+        typeMapping.put("UUID", "TEXT");
+        typeMapping.put("BigDecimal", "DECIMAL");
+
+        embeddedTemplateDir = templateDir = "mysql-schema-server";
+
+        // it seems that cli options from DefaultCodegen are useless here
+        cliOptions.clear();
+        addOption(DEFAULT_DATABASE_NAME, "Default database name for all MySQL queries", defaultDatabaseName);
+        addSwitch(JSON_DATA_TYPE_ENABLED, "Use special JSON MySQL data type for complex model properties. Requires MySQL version 5.7.8. Generates TEXT data type when disabled", jsonDataTypeEnabled);
+    }
+
+    @Override
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    @Override
+    public String getName() {
+        return "mysql-schema";
+    }
+
+    @Override
+    public String getHelp() {
+        return "Generates a MySQL schema.";
```

Comment:
What about "Generates a MySQL schema based on the model or schema defined in the OpenAPI specification (v2, v3)" ?